binding/flutter/android/src/main/java/ai/picovoice/flutter/porcupine/PorcupinePlugin.java

```diff
@@ -55,55 +53,80 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
     try {
       method = Method.valueOf(call.method.toUpperCase());
     } catch (IllegalArgumentException e) {
-      method = Method.DEFAULT;
+      result.error(
+              PorcupineRuntimeException.class.getSimpleName(),
+              String.format("Porcupine method '%s' is not a valid function", call.method),
+              null);
+      return;
     }
 
     switch (method) {
-      case INIT:
+      case FROM_BUILTIN_KEYWORDS:
         try {
-          // default model file
-          final File resourceDirectory = flutterContext.getFilesDir();
-          final Map<String, Object> constants = new HashMap<>();
-          constants.put("DEFAULT_MODEL_PATH", new File(resourceDirectory, "porcupine_params.pv").getAbsolutePath());
-
-          // default keyword files
-          final Map<String, String> keywordPaths = new HashMap<>();
-          for (Porcupine.BuiltInKeyword x : Porcupine.BuiltInKeyword.values()) {
-            String fileName = x.name().toLowerCase();
-            String keyword = fileName.replace('_', ' ');
-            keywordPaths.put(keyword, new File(resourceDirectory, fileName + ".ppn").getAbsolutePath());
+          String accessKey = call.argument("accessKey");
+          String modelPath = call.argument("modelPath");
+          ArrayList<String> keywordsList = call.argument("keywords");
+          ArrayList<Double> sensitivitiesList = call.argument("sensitivities");
+
+          Porcupine.BuiltInKeyword[] keywords = null;
+          if (keywordsList != null) {
+            keywords = new Porcupine.BuiltInKeyword[keywordsList.size()];
+            for (int i = 0; i < keywordsList.size(); i++) {
+              keywords[i] = Porcupine.BuiltInKeyword.valueOf(keywordsList.get(i));
```

Comment:
we should probably handle the case where we can't translate given keyword to built-in